server/api/reingest/reingest.controller.ts

```diff
@@ -0,0 +1,137 @@
+import { Response } from 'express';
+import { ApiRequest, EdipiParam } from '../index';
+import config from '../../config';
+import { dateFromString } from '../../util/util';
+import AWS from 'aws-sdk';
+import { BadRequestError, InternalServerError } from '../../util/error-types';
+
+class ReingestController {
+
+  static max_payload_size = 100;
+
+  async reingestSymptomRecords(req: ApiRequest<EdipiParam, TimestampData>, res: Response) {
+
+    let startDate = 0;
+    let endDate = Date.now();
+    if (req.body.startTime) {
+      startDate = ReingestController.convertDateParam(req.body.startTime);
+    }
+
+    if (req.body.endTime) {
+      endDate = ReingestController.convertDateParam(req.body.endTime);
+    }
+
+    const dynamoDB = new AWS.DynamoDB();
+    const queryInput = {
+      TableName: config.dynamo.symptomTable,
+      IndexName: config.dynamo.symptomIndex,
+      KeyConditions: {
+        "EDIPI": {
+          ComparisonOperator: "EQ",
+          AttributeValueList: [
+            {
+              S: req.params.edipi,
+            },
+          ],
+        },
+        "Timestamp": {
+          ComparisonOperator: "BETWEEN",
+          AttributeValueList: [
+            {
+              N: startDate.toString(),
+            },
+            {
+              N: endDate.toString(),
+            },
+          ],
+        },
+      },
+
+    };
+
+    try {
+      const data = await dynamoDB.query(queryInput).promise();
+
+      const {payloads, recordsIngested} = ReingestController.buildPayloadsFromData(data);
+
+      const lambda = new AWS.Lambda();
+      let lambdaInvocationCount = 0;
+      for (const lambdaPayload in payloads) {
+        const lambdaParams = {
+          FunctionName: config.dynamo.symptomLambda,
+          Payload: Buffer.from(JSON.stringify(lambdaPayload)),
+          InvocationType: 'Event'
+        }
+        lambda.invoke(lambdaParams);
+        lambdaInvocationCount++;
+      }
+
+      res.status(201).json({
+        'reingest-count': recordsIngested,
+        'lambda-invoke-count': lambdaInvocationCount
+      });
+    } catch(err) {
+      console.log('Reingest Failed: ', err.message);
+      throw new InternalServerError(`Error during reingest request: ${err.message}`);
+    }
+
+  }
+
+  static convertDateParam(date: string|number|Date): number {
+    if (typeof date == "string") {
+      const convertedDate = dateFromString(date);
+      if (!convertedDate) {
+        throw new BadRequestError(`Invalid date string provided for reingest: ${date}`);
+      }
+      return convertedDate.getTime();
+    } else if (date instanceof Date) {
+      return date.getTime();
+    }
+    return date
+  }
+
+  static buildPayloadsFromData(data: AWS.DynamoDB.QueryOutput) {
```

Comment:
I think these static functions should be moved out of the class, since they appear to just be internal helper functions that don't need to be exported.